storage/replica.go

```diff
@@ -191,43 +192,56 @@ func NewReplica(desc *roachpb.RangeDescriptor, store *Store) (*Replica, error) {
 		tsCache:     NewTimestampCache(store.Clock()),
 		sequence:    NewSequenceCache(desc.RangeID),
 		pendingCmds: map[cmdIDKey]*pendingCmd{},
+		RangeID:     desc.RangeID,
 	}
-	r.setDescWithoutProcessUpdate(desc)
 
-	lastIndex, err := r.loadLastIndex()
-	if err != nil {
+	if err := r.newReplicaInner(desc); err != nil {
 		return nil, err
 	}
+
+	if r.ContainsKey(keys.SystemDBSpan.Key) {
+		r.maybeGossipSystemConfig()
```

Comment:
If this called `maybeGossipSystemConfigLocked` it could be done inside `newReplicaInner`. And `newRangeStats` should also be callable under the lock, so I don't think there's a need for the helper function. Just do everything in `NewReplica` and defer the unlock.